integration-tests/tests/usage-reporting/advanced-breaking-changes.spec.ts

```diff
@@ -0,0 +1,123 @@
+import { initSeed } from 'testkit/seed';
+import { waitFor } from 'testkit/flow';
+
+describe('advanced breaking changes', async () => {
+  test('an argument can safely migrate from nullable to non nullable if all usages provide that argument', async () => {
+    const { createOrg } = await initSeed().createOwner();
+    const { createProject } = await createOrg();
+    const { createTargetAccessToken, updateTargetValidationSettings } = await createProject();
+    const { checkSchema, publishSchema, collectUsage } = await createTargetAccessToken({});
+
+    const userResult = await publishSchema({
+      sdl: /* GraphQL */ `
+        type Query {
+          user(id: ID): User
+          me: User
+        }
+
+        type User {
+          id: ID!
+          name: String
+        }
+      `,
+    });
+    await userResult.expectNoGraphQLErrors();
+
+    const usageReport = await collectUsage({
+      size: 2,
+      map: {
+        '4de79319': {
+          operationName: 'user',
+          operation: 'query user { user(id: "1234") { id } }',
+          fields: ['Query', 'Query.user', 'Query.user.id', 'User', 'User.id', 'ID'],
```

Comment:
Shouldn't there be a `Query.user.id!` being reported here? Can you elaborate on how this works?

Reply:
Turns out I was not waiting long enough after collecting usage.
I've fixed that and fixed a case with arguments that exists in the current version.
This issue is that if an argument is not provided, then its coordinate is not collected. And then the total count is off for that coordinate for the nullability case. I added a special condition for this to count the number of times the parent field is used when the change is from nullable to required.